src/slot_provider/mod.rs

```diff
@@ -1,53 +1,50 @@
 //! Slot providers provide a context which takes a fn, which is then called once for every slot.
-use chrono::{TimeZone, Utc};
-use eyre::Result;
-use futures::Future;
-use tokio::time::Duration;
+//! TODO: convert timestamp to DateTime
```

Comment:
In various places you turn a DateTime into a timestamp. I'd suggest using DateTime everywhere. Not only do you do less conversion, you don't have to worry about adding or subtracting from the wrong number. i.e. be mindful you're working in seconds. Total JS habit for me that is a breadth of fresh air in Rust. You can simply add seconds, or minutes, to a DateTime and it'll do the right thing.

Reply:
Yes good idea. The code could definetely use more consistency in datetime handling. Feel free to refactor this at will.